models/articles.models.js

```diff
@@ -26,6 +27,18 @@ function fetchArticles() {
     .then(response => {
         return response.rows;
     })
+} else {
+    return db
+    .query(`SELECT * FROM articles WHERE articles.topic = $1;`,
+    [topic])
+    .then((response) => {
+        if(response.rowCount === 0) {
+            return Promise.reject({status: 404, msg: "topic does not exist"})
```

Comment:
Getting no rows from this query doesn't necessarily mean the topic doesn't exist. 

Topics can exist but have 0 rows, think about the scenario when you first make a new topic. There'll be no articles associated with it for a while.